test/unit/lib/plugins/aws/provider.test.js

```diff
@@ -1216,6 +1217,40 @@ aws_secret_access_key = CUSTOMSECRET
         expect(spawnExtStub).to.be.calledWith('docker', ['push', `${repositoryUri}:baseimage`]);
       });
 
+      it('should work correctly when repository does not exist beforehand and scanOnPush is set', async () => {
+        const awsRequestStubMap = {
+          ...baseAwsRequestStubMap,
+          ECR: {
+            ...baseAwsRequestStubMap.ECR,
+            describeRepositories: describeRepositoriesStub.throws({
+              providerError: { code: 'RepositoryNotFoundException' },
+            }),
+            createRepository: createRepositoryStubScanOnPush.resolves({
+              repository: { repositoryUri },
+            }),
+          },
+        };
+
+        const { awsNaming, cfTemplate } = await runServerless({
```

Comment:
Awesome job with the test :clap: 

As for the fixture, I think we don't need a new one - essentially we try to reuse fixtures if possible and don't create new ones that are used in only one example. Here we could stick to `ecr` fixture and add additional config via `configExt` property, in a similar way as it's done here: https://github.com/serverless/serverless/blob/6d2ed840fb21d469d8e4a548a38ca9302d5a9d2c/test/unit/lib/plugins/aws/provider.test.js#L862-L878

Reply:
done